src/script/miniscript.h

```diff
@@ -1139,24 +1314,37 @@ struct Node {
     //! Return the number of ops in the script (not counting the dynamic ones that depend on execution).
     uint32_t GetStaticOps() const { return ops.count; }
 
-    //! Check the ops limit of this script against the consensus limit.
-    bool CheckOpsLimit() const { return GetOps() <= MAX_OPS_PER_SCRIPT; }
+    //! Check the ops limit of this script against the consensus limit. This limit is only present for P2WSH.
+    template <typename Ctx> bool CheckOpsLimit(const Ctx& ctx) const {
+        return ctx.MsContext() == MiniscriptContext::TAPSCRIPT || GetOps() <= MAX_OPS_PER_SCRIPT;
+    }
+
+    //! Return the maximum number of stack elements needed to satisfy this script non-malleably.
+    uint32_t GetStackSize() const { return ss.sat.size.value; }
 
-    /** Return the maximum number of stack elements needed to satisfy this script non-malleably, including
-     * the script push. */
-    uint32_t GetStackSize() const { return ss.sat.value + 1; }
+    //! Return the maximum size of the stack during execution of this script.
+    uint32_t GetExecStackSize() const {
+        return ss.sat.size.value + (uint32_t)std::max(ss.sat.exec.value, 0);
+    }
 
-    //! Check the maximum stack size for this script against the policy limit.
-    bool CheckStackSize() const { return GetStackSize() - 1 <= MAX_STANDARD_P2WSH_STACK_ITEMS; }
+    //! Check the maximum stack size for this script against the limits.
+    template <typename Ctx> bool CheckStackSize(const Ctx& ctx) const {
+        if (ctx.MsContext() == MiniscriptContext::P2WSH && GetStackSize() > MAX_STANDARD_P2WSH_STACK_ITEMS) {
+            return false;
+        }
```

Comment:
Why not this?
```C++
        if (ctx.MsContext() == MiniscriptContext::P2WSH) {
            return GetStackSize() <= MAX_STANDARD_P2WSH_STACK_ITEMS;
        }
```
Otherwise, it seems to add the other condition for the `P2WSH` context as well.

Reply:
Yeah, it didn't matter since the computation is trivial and the bounds more loose for Tapscript anyways. But your suggestion is clearer indeed, taken, thanks.